test/cache_test.lua

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env tarantool
+
+local t = require('luatest')
+local db = require('test.db')
+local ddl = require('ddl')
+local cache = require('ddl.cache')
+local helper = require('test.helper')
+
+local SPACE_NAME_IDX      = 1
+local SHARD_FUNC_NAME_IDX = 2
+local SHARD_FUNC_BODY_IDX = 3
+
+local primary_index = {
+    type = 'HASH',
+    unique = true,
+    parts = {
+        {path = 'string_nonnull', is_nullable = false, type = 'string'},
+        {path = 'unsigned_nonnull', is_nullable = false, type = 'unsigned'},
+    },
+    name = 'primary'
+}
+
+local bucket_id_idx = {
+    type = 'TREE',
+    unique = false,
+    parts = {{path = 'bucket_id', type = 'unsigned', is_nullable = false}},
+    name = 'bucket_id'
+}
+
+local func_body_first  = 'function() return 42 end'
+local func_body_second = 'function() return 24 end'
+
+local function rebuild_db(g)
+    db.drop_all()
+
+    g.space = {
+        engine = 'memtx',
+        is_local = true,
+        temporary = false,
+        format = table.deepcopy(helper.test_space_format())
+    }
+    table.insert(g.space.format, 1, {
+        name = 'bucket_id', type = 'unsigned', is_nullable = false
+    })
+
+    g.space.indexes = {
+        table.deepcopy(primary_index),
+        table.deepcopy(bucket_id_idx)
+    }
+    g.space.sharding_key = {'unsigned_nonnull', 'integer_nonnull'}
+    g.schema = {
+        spaces = {
+            space = g.space,
+        }
+    }
+end
+
+local g = t.group()
+g.before_all(db.init)
+g.before_each(function()
+    rebuild_db(g)
+end)
```

Comment:
Usually test setup used for initialization and other steps required for test and teardown used to cleanup test data or destroy test data. You are using  both in `rebuild_db` and call it in `before_each`, function for test setup. What is a point of this?

Can we do like this:

```lua

local function space_init(g)
    g.space = {
        engine = 'memtx',
        is_local = true,
        temporary = false,
        format = table.deepcopy(helper.test_space_format())
    }
    table.insert(g.space.format, 1, {
        name = 'bucket_id', type = 'unsigned', is_nullable = false
    })

    g.space.indexes = {
        table.deepcopy(primary_index),
        table.deepcopy(bucket_id_idx)
    }
    g.space.sharding_key = {'unsigned_nonnull', 'integer_nonnull'}
    g.schema = {
        spaces = {
            space = g.space,
        }
    }
end

g.before_each() = space_init()

g.after_each = db.drop_all()
```

With such setup and teardown, it is clear what  happen before and after each test.

Reply:
Thanks for the tip! I will use this method.

Reply:
fixed

Reply:
--g.after_each = db.drop_all()
we need to clear db before first test call also, db is dirty after other tests.

Reply:
(It seems, it is not actual anymore. Resolving the discussion.)